iavl/internal/mmap.go

```diff
@@ -0,0 +1,79 @@
+package internal
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+import "github.com/edsrzf/mmap-go"
```

Comment:
For now I am using this off-the-shelf mmap wrapper which has the highest number of known importers on  pkg.go.dev: https://pkg.go.dev/github.com/edsrzf/mmap-go?tab=importedby

In the future, it may be worth considering creating our own mmap wrapper. On linux, it may be possible to apply an optimization where we can resize the mmap without unmapping memory: https://stackoverflow.com/questions/74243583/memory-map-file-with-growing-size